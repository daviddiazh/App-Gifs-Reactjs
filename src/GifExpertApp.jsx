import React from 'react'

export const GifExpertApp = () => {
    return (
        <>
            <p>Hola mundo</p>
        </>
    )
}
