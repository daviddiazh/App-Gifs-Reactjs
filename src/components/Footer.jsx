import React from 'react'

export const Footer = () => {

    const year = new Date().getFullYear();

    return (
        <footer>
            <p>&copy; Powered by. David Diaz H | {year}.</p>
        </footer>
    )
}
