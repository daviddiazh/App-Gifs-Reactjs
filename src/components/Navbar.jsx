import React from 'react'
import gif from '../images/gif.png'

export const Navbar = () => {
    return (
        <nav className='navbar'>
            <h1 style={{ marginTop: '18px' }}>GifsApp</h1>
            <img className='img-gif' src={gif} width={100} alt='Gif Image' />
        </nav>
    )
}
