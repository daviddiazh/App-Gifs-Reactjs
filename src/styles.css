* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
}

body {
  padding: 50px;
}

form {
  margin: 0 auto;
  text-align: center;
}
  
input {
  background-color: #fff;
  border-radius: 6px;
  /* border: 1px solid rgb(97, 32, 158); */
  border: none;
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 15px;
  width: 65%;
}
  
h2 {
  font-size: 1.5rem;
}
  
h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}

footer p {
  text-align: center;
}

.navbar {
  display: flex;
  justify-content: space-between;
}

.img-gif {
  width: 35px;
  height: 35px;
  padding: 10px 0 0 0;
}
  
.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
  
.card {
  align-content: center;
  align-items: center;
  background-color: #fff;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;  
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
}
  
.card p {
  background-color: #fff;
  flex: 1;
  font-size: 1rem;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
}
  
.card img {
  width: 100%;
}

.example-search {
  color: #7f7b7b;
  margin: 7px 2px;
}

@media (max-width: 800px) {
  input {
    width: 97%;
    margin: 0;
    padding: 12px 5px;
  }
}