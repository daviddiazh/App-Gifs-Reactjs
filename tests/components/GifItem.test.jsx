import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";


describe('Testing of <GifItem />', () => {

    const title = 'Hello World';
    const url = 'https://image.com.co';
    //? For look as console.log, use screen.debug();

    test('Should create a snapshot', () => {
        const { cotainer } = render(<GifItem title={title} url={url} />);
        expect( cotainer ).toMatchSnapshot(); 
    });

    test('Should look the image with URL and property ALT ', () => {
        render(<GifItem title={title} url={url} />);
        //screen.debug();
        //expect( screen.getByRole('img').src ).toBe(url);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

});