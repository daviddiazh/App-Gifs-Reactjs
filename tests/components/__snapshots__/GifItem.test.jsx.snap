// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of <GifItem /> Should create a snapshot 1`] = `undefined`;
